packages/core/src/types.ts

```diff
@@ -359,11 +376,30 @@ export type ExtractRematchDispatcherFromReducer<
  * Otherwise, it describes dispatcher which accepts one argument (payload)
  * and returns an action.
  */
-export type RematchDispatcher<TPayload = void> = [TPayload] extends [void]
-	? (() => Action<void>) & { isEffect: false }
-	: undefined extends TPayload
-	? ((payload?: TPayload) => Action<TPayload>) & { isEffect: false }
-	: ((payload: TPayload) => Action<TPayload>) & { isEffect: false }
+export type RematchDispatcher<TPayload = void, TMeta = void> = [
```

Comment:
I added the type support for optional `meta` in `RematchDispatcher` and `ExtractRematchDispatcherFromReducer`.